scripting/src/main/java/org/infinispan/scripting/impl/ScriptingManagerImpl.java

```diff
@@ -165,10 +165,13 @@ public boolean containsScript(String taskName) {
          .orElseGet(() -> new SimpleBindings());
 
       SimpleBindings systemBindings = new SimpleBindings();
-      systemBindings.put(SystemBindings.CACHE_MANAGER.toString(), cacheManager);
+      DataTypedCacheManager cm = new DataTypedCacheManager(metadata.dataType(), context.getMarshaller(), cacheManager);
+      systemBindings.put(SystemBindings.CACHE_MANAGER.toString(), cm);
       systemBindings.put(SystemBindings.SCRIPTING_MANAGER.toString(), this);
       context.getCache().ifPresent(cache -> {
-         systemBindings.put(SystemBindings.CACHE.toString(), cache);
+         Cache<?, ?> c = cm.getCacheConfiguration(cache.getName()).compatibility().enabled()
```

Comment:
This calls `cm.getCacheConfiguration(cache.getName()).compatibility().enabled()` instead of `cache.getCacheConfiguration().compatibility().enabled()` because the latter requires ADMIN rights in secured caches whereas the former works.

The bigger problem is that even if the cache is secured, we should be able to call what we need to call without problems. Maybe this can be implemented in a better way?